instrumentation/mongo/mongo-4.0/javaagent/mongo-4.0-javaagent.gradle

```diff
@@ -0,0 +1,26 @@
+apply from: "$rootDir/gradle/instrumentation.gradle"
+
+muzzle {
+  pass {
+    group = "org.mongodb"
+    module = "mongodb-driver-core"
+    // this instrumentation is backwards compatible with early versions of the new API that shipped in 3.7
+    // the legacy API instrumented in mongo-3.1 continues to be shipped in 4.x, but doesn't conflict here
+    // because they are triggered by different types: MongoClientSettings(new) vs MongoClientOptions(legacy)
+    versions = "[3.7,)"
+    assertInverse = true
```

Comment:
but  before 4.0, non-uber jars have also been published, that's why 3.7 is here

Reply:
I don't follow, sorry. If this module is meant for mongo 4.0+, then why do we want to verify muzzle for 3.7?

Reply:
there are 2 types of mongo jars:
1. uber jar. But it doesn't exist since version 4.0. 
2. non-uber jar which exists since version 3.7 up to current version.


DD named module to support non-uber jar `mongo-4`. 
Should I rename it to `mongo-non-uber-3.7`?

Reply:
Wait, so actual instrumentations are the same, right? The only difference is, essentially, the collection of packages used for testing? So this is not a new module for instrumentation, but a new module for testing? Like `glassfish-testing`?

Reply:
It's an actual instrumentation for non-uber jars.

Reply:
How does it differ from already existing `io.opentelemetry.javaagent.instrumentation.mongo.v3_7.MongoClientInstrumentationModule`? Instrumentation looks the same to me.

Reply:
yes, instrumentation looks the same.
but mongo-3.7 will not work for `mongodb-driver-core` 3.7 jar because of muzzle.  
`mongo-3.7` depends on `mongo-java-driver`
`mongo-4.0` depends on `mongodb-driver-core`

Reply:
Instead of copying the 3.7 instrumentation, would it be possible to add another muzzle task there and just move the tests to a separate 4.0/uberjar testing module?

Reply:
Adding

```groovy
  pass {
    group = "org.mongodb"
    module = "mongodb-driver-core"
    // this instrumentation is backwards compatible with early versions of the new API that shipped in 3.7
    // the legacy API instrumented in mongo-3.1 continues to be shipped in 4.x, but doesn't conflict here
    // because they are triggered by different types: MongoClientSettings(new) vs MongoClientOptions(legacy)
    versions = "[3.7,)"
    extraDependency "org.mongodb:bson:3.7.0"
    assertInverse = true
  }
```

to the 3.7 muzzle task definition should cover all the uberjar releases.

Reply:
@mateuszrzeszutek done in https://github.com/open-telemetry/opentelemetry-java-instrumentation/pull/2046/commits/5e5ea23168589a266496e8d5b2e3d1461651a558